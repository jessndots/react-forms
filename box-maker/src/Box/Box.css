.Box {
  display: flex;
}

.Box-Box {
  background-color: white;
  height: 100px;
  width: 100px;
  border: 1px solid black;
  margin: 2px;
}

.Box-X {
  height: 50%;
}