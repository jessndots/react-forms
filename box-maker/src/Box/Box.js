import React from "react";
import './Box.css'


const Box = ({ backgroundColor, width, height, id, deleteBox }) => {
    const divStyle = {
        backgroundColor: backgroundColor,
        height: height + 'px',
        width: width + 'px'
    }
    return (
        <div className="Box" id={id}>
            <button className="Box-X" onClick={evt => deleteBox(id)}>X</button>
            <div
                className="Box-Box"
                style={divStyle}
            >
            </div>
        </div>

    )
}

export default Box;