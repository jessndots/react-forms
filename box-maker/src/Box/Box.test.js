import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

// Smoke test
it("renders without crashing", function() {
  render(<Box/>)
})

// Snapshot test
it("matches screenshot", function() {
  const {asFragment} = render(<Box/>);
  expect(asFragment()).toMatchSnapshot();
})