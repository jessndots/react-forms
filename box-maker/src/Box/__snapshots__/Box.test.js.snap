// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <button
      class="Box-X"
    >
      X
    </button>
    <div
      class="Box-Box"
    />
  </div>
</DocumentFragment>
`;
